clang/test/SemaSYCL/intel-fpga-local-ast.cpp

```diff
@@ -0,0 +1,537 @@
+  //CHECK: SYCLIntelMaxReplicatesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::max_replicates(12)]]
+  [[intel::max_replicates(12)]] int var_max_replicates; // OK
+
+  // Check duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_private_copies
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelPrivateCopiesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::private_copies(12)]]
+  [[intel::private_copies(12)]] int var_private_copies; // OK
+
+  // Checking of duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_forcep2d
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}1
+  //CHECK-NEXT: IntegerLiteral{{.*}}1{{$}}
+  [[intel::force_pow2_depth(1)]]
+  [[intel::force_pow2_depth(1)]] int var_forcep2d; // OK
+
+  // Test for Intel FPGA bankwidth memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_bankwidth 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 16
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 16
+  [[intel::bankwidth(16)]]
+  [[intel::bankwidth(16)]] int var_bankwidth; // OK
+
+  // Test for Intel FPGA numbanks memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_numbanks 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  [[intel::numbanks(8)]]
+  [[intel::numbanks(8)]] int var_numbanks; // OK
+
+  // Checking of different argument values.
+  //CHECK: VarDecl{{.*}}bw_bw 'unsigned int[64]'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //expected-warning@+2{{attribute 'bankwidth' is already applied}}
+  [[intel::bankwidth(8)]] // expected-note {{previous attribute is here}}
+  [[intel::bankwidth(16)]] unsigned int bw_bw[64];
+
+  //CHECK: VarDecl{{.*}}nb_nb 'unsigned int[64]'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //expected-warning@+2{{attribute 'numbanks' is already applied}}
+  [[intel::numbanks(8)]] // expected-note {{previous attribute is here}}
+  [[intel::numbanks(16)]] unsigned int nb_nb[64];
+
+  //Last one is applied and others ignored.
```

Comment:
It does not look like 'others' are ignored. I see both merge attributes being checked in CHECK here. Doesn't that mean both are applied?

Reply:
Thanks @elizabethandrews for reviews! Yes, both merge attributes are applied here. This is an existing test. I just copied it. I have created https://github.com/intel/llvm/issues/7818 to investigate and address the problem in a separate patch since i am not making any functional change here.